assets/app/view/spreadsheet.rb

```diff
@@ -31,15 +31,44 @@ def render_table
       # TODO: consider adding train availability
     end
 
+    def or_history(corporations)
+      # It looks like the set class isn't available, so we use this as a proxy
+      or_hist = {}
+      corporations.each do |c|
+        c.revenue_history.keys.each { |x| or_hist[x] = true }
+      end
```

Comment:
ok i see what this is doing now, a bit convoluted, you can just do

corporations.flat_map { |c| c.revenue_history.keys }.uniq.sort

Reply:
Ah Ruby. "uniq". Of course. My Google Fu failed me earlier -- thanks!